lib/elixir/lib/calendar.ex

```diff
@@ -151,11 +244,18 @@ defmodule Date do
       true
 
   """
-  @spec utc_today() :: t
-  def utc_today() do
+  @spec utc_today(Calendar.calendar) :: t
+  def utc_today(calendar \\ Calendar.ISO)
+  def utc_today(Calendar.ISO) do
     {:ok, {year, month, day}, _, _} = Calendar.ISO.from_unix(:os.system_time, :native)
     %Date{year: year, month: month, day: day}
   end
+  def utc_today(calendar) do
+    {:ok, {year, month, day}, _, _} = Calendar.ISO.from_unix(:os.system_time, :native)
+    %Date{year: year, month: month, day: day, calendar: Calendar.ISO}
```

Comment:
Because `Date.convert/2` will return an error tuple if the two calendars are not compatible (`Calendar.compatible_calendars?/2`). This check is skipped by doing it inline. We are allowed to skip this because `utc_today/1` will use midnight as time pivot to do the conversion.

However, thinking about it a little more I am not entirely sure that this is the correct approach. It's also possible to make an argument for doing the conversion on the DateTime fields that `Calendar.ISO.from_unix` returns before changing them to Dates. (This would then be similar to the original version I had which used `DateTime.convert |> DateTime.to_date`)

Hmm...